src/sensors/sonar.py

```diff
@@ -64,58 +75,81 @@ def draw(self):
                     square_coords = (x * cell_size, y * cell_size,
                                      x * cell_size, y * cell_size + cell_size,
                                      x * cell_size + cell_size, y * cell_size,
-                                     x * cell_size + cell_size, y * cell_size + cell_size)
+                                     x * cell_size + cell_size,
+                                     y * cell_size + cell_size)
                     pyglet.graphics.draw_indexed(4, pyglet.gl.GL_TRIANGLES,
                                                  [0, 1, 2, 1, 2, 3],
                                                  ('v2i', square_coords))
 
 
 class Sonar(object):
-    def __init__(self, sensor_map, min_range, max_range, cone_angle):
+    """
+    TODO:
+    - add gaussian noise / variance
+    - use OpenGL to draw the rays
+    - Use intensity to simulate min/max range.
+    """
+
+    def __init__(self, sensor_map, min_range, max_range, cone_angle, beams=30):
         self.min_range = min_range
         self.max_range = max_range
         self.cone_angle = cone_angle
         self.sensor_map = sensor_map
+        self.beams = beams
         self.current_range = -1.0
 
     def update_sonar(self, x, y, theta):

```

Comment:
Can we link this to the getDistance() method in the API for the two robots?  That way it is only called when getDistance() is explicitly invoked by the user/programmer.  This will make everything much more efficient and prevent other functions being held up because update_sonar is constantly executing.

Reply:
Just leaving a note to remind myself and inform others - that this doesn't work in the simulator.  The getDistance method is on the "python program" with no direct link to the simulated robot.  The robot's overall state on the python program side is updated by reading from a socket link that the simulated robot publishes to regularly.  We would need to rework the entire communication pattern to have distance only calculated on request.